mobile/src/main/java/org/openhab/habdroid/ui/homescreenwidget/VoiceWidget.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2010-2016, openHAB.org and others.
+ *
+ *   All rights reserved. This program and the accompanying materials
+ *   are made available under the terms of the Eclipse Public License v1.0
+ *   which accompanies this distribution, and is available at
+ *   http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.habdroid.ui.homescreenwidget;
+
+import android.app.PendingIntent;
+import android.appwidget.AppWidgetManager;
+import android.appwidget.AppWidgetProvider;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.speech.RecognizerIntent;
+import android.speech.SpeechRecognizer;
+import android.util.Log;
+import android.widget.RemoteViews;
+import androidx.annotation.LayoutRes;
+
+import org.openhab.habdroid.R;
+import org.openhab.habdroid.core.VoiceService;
+
+/**
+ * Implementation of App Widget functionality.
+ */
+public class VoiceWidget extends AppWidgetProvider {
```

Comment:
@maniac103 Should this be abstract with two subclasses that both override setupOpenhabIcon() and getLayoutRes()?